pkg/common/cns-lib/volume/manager.go

```diff
@@ -1014,3 +1016,62 @@ func (m *defaultManager) RetrieveVStorageObject(ctx context.Context, volumeID st
 	log.Debugf("vStorageObject for volumeID: %q is %+v", volumeID, vStorageObject)
 	return vStorageObject, nil
 }
+
+// QueryVolumeAsync returns volumes matching the given filter using CnsQueryAsync API.
+func (m *defaultManager) QueryVolumeAsync(ctx context.Context, queryFilter cnstypes.CnsQueryFilter, querySelection cnstypes.CnsQuerySelection) (*cnstypes.CnsQueryResult, error) {
+	internalQueryVolume := func() (*cnstypes.CnsQueryResult, error) {
+		log := logger.GetLogger(ctx)
+		err := validateManager(ctx, m)
+		if err != nil {
+			return nil, err
+		}
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		// Set up the VC connection
+		err = m.virtualCenter.ConnectCns(ctx)
+		if err != nil {
+			log.Errorf("ConnectCns failed with err: %+v", err)
+			return nil, err
+		}
+		//Call the CNS QueryVolume
+		res, err := m.virtualCenter.CnsClient.QueryVolume(ctx, queryFilter)
```

Comment:
Why do we need to call QueryVolume here?

Reply:
My bad, must have been a copy paste error